packages/caver-klay/caver-klay-accounts/src/index.js

```diff
@@ -800,47 +826,82 @@ Accounts.prototype.decrypt = function (v3Keystore, password, nonStrict) {
     
     var json = (_.isObject(v3Keystore)) ? v3Keystore : JSON.parse(nonStrict ? v3Keystore.toLowerCase() : v3Keystore);
 
-    if (json.version !== 3) {
-        console.warn('This is not a V3 wallet.')
+    if (json.version !== 3 && json.version !== 4) {
+        console.warn('This is not a V3 or V4 wallet.')
         // throw new Error('Not a valid V3 wallet');
     }
 
-    var derivedKey;
-    var kdfparams;
-    /**
-     * Supported kdf modules are the following:
-     * 1) pbkdf2
-     * 2) scrypt
-     */
-    if (json.crypto.kdf === 'scrypt') {
-        kdfparams = json.crypto.kdfparams;
+    if (json.version === 3 && !json.crypto) {
+      // crypto field should be existed in keystore version 3
+      throw new Error(`Invalid keystore V3 format: 'crypto' is not defined.`)
+    }
+
+    if (json.crypto) {
+      if (json.keyRing) throw new Error(`Invalid key store format: 'crypto' can not be with 'keyRing'`)
+      json.keyRing = [[json.crypto]]
+      delete json.crypto
+    }
 
-        // FIXME: support progress reporting callback
-        derivedKey = scrypt(Buffer.from(password), Buffer.from(kdfparams.salt, 'hex'), kdfparams.n, kdfparams.r, kdfparams.p, kdfparams.dklen);
-    } else if (json.crypto.kdf === 'pbkdf2') {
-        kdfparams = json.crypto.kdfparams;
+    if (json.keyRing.length !== 1 && json.keyRing.length !== 3) throw new Error(`Invalid key store format`)
```

Comment:
I think length 2 is also possible if the user wants to set transaction and accountUpdate role only.

Reply:
Um, i thought if user wants to set transaction and accountUpdate role only, the keysotre format will be like [ [{}], [{}], [] ].

Reply:
Hmm... As you might know or not, we have a chance that another role could be introduced in the future. In that case, how can we express that? And, how can we decode three roles (current) and four roles (future) together?

Reply:
Um, okay. 
To be flexible as new roles are added, it seems right to implement them as you say. 
If so, it may be necessary to change the format i mentioned yesterday about format.

Reply:
The implementation for modification format of keystore is applied.

Now role based format like below
1. transactionKey, feePayerKey
[ [ {}, {} ], [], [ {} ] ] 
2. transactionKey, updateKey
[ [ {}, {} ], [ {}, {} ] ]
3. feePayerKey
[ [], [], [ {} ] ]
4. transactionKey
[ [ {}, {} ] ]

Now MultiSig or public Key format like below
1. Multiple private key strings
[ {}, {}, {} ]
2. single private key
[ {} ]